### PR TITLE
Expose managesieve to the outside world

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

Feature

## What does this PR do?

Expose managesieve to the outside world.

### Related issue(s)
- #428
- #113
- #81

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [ ] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
It's a little bit quirky right now:
- only implicit TLS is supported (should we use the standard port?)
- dovecot doesn't know about the real client IP
```
c5ab2744bdbc:/app# doveadm who
username          # proto (pids) (ips)                                           
admin@example.com 1 sieve (454)  (192.168.203.8)
```
- authentication never fails permanently; we always return a temporary error:
```
"IMPLEMENTATION" "Dovecot Pigeonhole"
"SIEVE" "fileinto reject envelope encoded-character vacation subaddress comparator-i;ascii-numeric relational regex imap4flags copy include variables body enotify environment mailbox date index ihave duplicate mime foreverypart extracttext"
"NOTIFY" "mailto"
"SASL" "PLAIN LOGIN"
"VERSION" "1.0"
OK "Dovecot ready."
AUTHENTICATE "PLAIN" "AGFkbWluQGV4YW1wbGUuY29tAGxldG1laW4y"
NO (TRYLATER) "Temporary authentication failure. [c5ab2744bdbc:2023-04-14 17:02:45]"
```

@fastlorenzo let me know what you think.

**Comment by nextgens** [comment]
The temporary auth fail is now solved; we can adjust the message as required.
```
"IMPLEMENTATION" "Dovecot Pigeonhole"
"SIEVE" "fileinto reject envelope encoded-character vacation subaddress comparator-i;ascii-numeric relational regex imap4flags copy include variables body enotify environment mailbox date index ihave duplicate mime foreverypart extracttext"
"NOTIFY" "mailto"
"SASL" "PLAIN LOGIN"
"VERSION" "1.0"
OK "Dovecot ready."
AUTHENTICATE "PLAIN" "AGFkbWluQGV4YW1wbGUuY29tAGxldG1laW4y"
NO "AuthFailed"
```

**Comment by fastlorenzo** [comment]
Will test it out asap

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Review by Diman0 — DISMISSED** [review]
- Changelog entry is missing.
- Proxy protocol documentation in Configuration.rst must be updated with the new `sieve` value.

Does the rate limiting functionality also work for managesieve? It will log the source ip address that is rate limited?

**Comment by nextgens** [comment]
> Does the rate limiting functionality also work for managesieve?

No, it won't if we implement it the way currently proposed in this PR.

File: docs/configuration.rst [hunk 1]
```diff
@@ -260,6 +260,7 @@ It can be set to:
 
 * ``http`` to accept the ``PROXY`` protocol on nginx's HTTP proxy ports
 * ``mail`` to accept the ``PROXY`` protocol on nginx's mail proxy ports
+* ``sieve`` to accept the ``PROXY`` protocol on nginx's managesieve proxy port
```
Comment (OdyX) [inline docs/configuration.rst:263]: Shall I suggest merging that with `mail` ? Having two made configuration easy (all combinations from "none, only mail, only http, both" are configurable).

With a third option, you either need 7 option names (to allow "mail + sieve" or "http + sieve", or complex string comparison.

**Comment by fastlorenzo** [comment]
@nextgens it doesn't seem to work for me for an obscure reason...

Here are packet captures of front and dovecot containers when accessing sieve from the internet:
image


I can confirm connectivity between front and dovecot works on port 4190: 
image

I've checked the front logs but I don't see anything logged when I attempt to reach sieve from the outside world.

File: docs/configuration.rst (hunk 1 above)
Comment (nextgens) [reply docs/configuration.rst]: Ok, changed in 801e6fe0

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
I think that #2790 is a better approach; I'll close this.

**Pull request closed without merging**